### PR TITLE
[frontend] define a default dashboard for a group or an organization (#4178)

### LINKED ISSUE #4178: Unable to define a default dashboard for a group or an organization
## Description

Go in Settings => Security => Groups => A group

Edit the group and set a default dashboard => No effect

Same behaviour in Organizations

### PR DESCRIPTION
**Pull request by yassine-ouaamou**
### Proposed changes

* Ability de select a default dashboard for a group or an organization

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/4178

### CODE REVIEW

**Comment by Archidoit** [comment]
Missing translations:
- Default dashboard
- tooltip content

image

**Comment by Archidoit** [comment]
behavior OK 💯

File: opencti-platform/opencti-front/src/private/components/settings/groups/GroupEditionOverview.tsx [hunk 1]
```diff
@@ -34,13 +36,38 @@ const groupEditionOverviewFocus = graphql`
...
 `;
 
-const groupValidation = (t: (value: string) => string) => Yup.object().shape({
-  name: Yup.string().required(t('This field is required')),
-  description: Yup.string().nullable(),
-  default_assignation: Yup.bool(),
-  auto_new_marking: Yup.bool(),
-  default_dashboard: Yup.object().nullable(),
-});
+const groupMutationRelationAdd = graphql`
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/settings/groups/GroupEditionOverview.tsx:39]: It's not a good idea to add these mutations, just to satisfy useFormEditor contracts.

We need to have a useFormEditor for internal object or have a composition of hooks.
@Kedae any ideas on it ?

I saw the same problem in SettingsOrganizationEdition

**Review by RomuDeuxfois — CHANGES_REQUESTED** [review]

File: opencti-platform/opencti-front/src/private/components/settings/groups/GroupEditionOverview.tsx (hunk 1 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/settings/groups/GroupEditionOverview.tsx:39]: Couldn't agree more.

**Review by SarahBocognano — COMMENTED** [review]
Tested ✅ Behavior works well, as @Archidoit said, the translation is missing but the rest is okay

File: opencti-platform/opencti-front/src/private/components/settings/groups/GroupEditionOverview.tsx (hunk 1 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/settings/groups/GroupEditionOverview.tsx:39]: An idea

```
import { GraphQLTaggedNode } from 'relay-runtime/lib/query/RelayModernGraphQLTag';
import { useMutation } from 'react-relay';
import { ObjectSchema, SchemaObjectDescription } from 'yup';
import { Option } from '../../private/components/common/form/ReferenceField';
import {
  convertAssignees,
  convertExternalReferences,
  convertKillChainPhases,
  convertMarkings, convertParticipants,
} from '../edition';

export interface GenericData {
  id: string;
  readonly objectMarking: {
    readonly edges: ReadonlyArray<{
      readonly node: {
        readonly definition: string | null;
        readonly definition_type: string | null;
        readonly id: string;
      };
    }>;
  } | null;
  readonly objectAssignee?: {
    readonly edges: ReadonlyArray<{
      readonly node: {
        readonly entity_type: string;
        readonly id: string;
        readonly name: string;
      };
    }>;
  } | null;
  readonly killChainPhases?: {
    readonly edges: ReadonlyArray<{
      readonly node: {
        readonly id: string;
        readonly kill_chain_name: string;
        readonly phase_name: string;
        readonly x_opencti_order: number | null;
      };
    }>;
  } | null;
}

interface Queries {
  fieldPatch: GraphQLTaggedNode;
  editionFocus: GraphQLTaggedNode;
  relationAdd?: GraphQLTaggedNode;
  relationDelete?: GraphQLTaggedNode;
}

const useFormEditor = (
  data: GenericData,
  enableReferences: boolean,
  queries: Queries,
  validator: ObjectSchema<{ [p: string]: unknown }>,
) => {
  const [commitFieldPatch] = useMutation(queries.fieldPatch);
  const [commitEditionFocus] = useMutation(queries.editionFocus);
  const schemaFields = (validator.describe() as SchemaObjectDescription).fields;

  const validate = (
    name: string,
    values: number | number[] | string | Date | Option | Option[],
    callback: () => void,
  ) => {
    if (schemaFields[name]) {
      validator
        .validateAt(name, { [name]: values })
        .then(() => {
          callback();
        })
        .catch(() => false);
    } else {
      callback();
    }
  };

  // Simple
  const changeCreated = (name: string, value: Option) => {
    if (!enableReferences) {
      validate(name, value, () => {
        commitFieldPatch({
          variables: {
            id: data.id,
            input: [
              {
                key: 'createdBy',
                value: [value.value],
              },
            ],
          },
        });
      });
    }
  };
  const changeFocus = (name: string) => {
    commitEditionFocus({
      variables: {
        id: data.id,
        input: {
          focusOn: name,
        },
      },
    });
  };
  const changeField = (name: string, value: number | number[] | string | Date | Option | Option[]) => {
    if (!enableReferences) {
      let finalValue = value;
      if (name === 'x_opencti_workflow_id') {
        finalValue = (value as Option).value;
      }
      validate(name, value, () => {
        commitFieldPatch({
          variables: {
            id: data.id,
            input: [{ key: name, value: finalValue || '' }],
          },
        });
      });
    }
  };

  // Multiple
  if (queries.relationAdd && queries.relationDelete) {
    const [commitRelationAdd] = useMutation(queries.relationAdd);
    const [commitRelationDelete] = useMutation(queries.relationDelete);
    const changeMultiple = (
      name: string,
      values: Option[],
      relation: string,
      optionMapper: (data: unknown) => [Option],
    ) => {
      if (!enableReferences) {
        validate(name, values, () => {
          const currentValues: [Option] = optionMapper(data);
          const added = values.filter(
            (v) => !currentValues.map((c) => c.value).includes(v.value),
          );
          const removed = currentValues.filter(
            (c) => !values.map((v) => v.value).includes(c.value),
          );
          if (added.length > 0) {
            commitRelationAdd({
              variables: {
                id: data.id,
                input: {
                  toId: added[0].value,
                  relationship_type: relation,
                },
              },
            });
          }
          if (removed.length > 0) {
            commitRelationDelete({
              variables: {
                id: data.id,
                toId: removed[0].value,
                relationship_type: relation,
              },
            });
          }
        });
      }
    };
    const changeMarking = (name: string, values: Option[]) => {
      changeMultiple(name, values, 'object-marking', convertMarkings);
    };
    const changeAssignee = (name: string, values: Option[]) => {
      changeMultiple(name, values, 'object-assignee', convertAssignees);
    };
    const changeParticipant = (name: string, values: Option[]) => {
      changeMultiple(name, values, 'object-participant', convertParticipants);
    };
    const changeKillChainPhases = (name: string, values: Option[]) => {
      changeMultiple(name, values, 'kill-chain-phase', convertKillChainPhases);
    };
    const changeExternalReferences = (name: string, values: Option[]) => {
      changeMultiple(
        name,
        values,
        'external-reference',
        convertExternalReferences,
      );
    };
    return {
      changeMarking,
      changeAssignee,
      changeParticipant,
      changeCreated,
      changeKillChainPhases,
      changeExternalReferences,
      changeFocus,
      changeField,
      fieldPatch: commitFieldPatch,
    };
  }

  return {
    changeCreated,
    changeFocus,
    changeField,
    fieldPatch: commitFieldPatch,
  };
};

export default useFormEditor;

```

**Comment by SamuelHassine** [comment]
@yassine-ouaamou TODO: please think about fixing comments from @RomuDeuxfois and @Kedae in another PR.

**Pull request merged**